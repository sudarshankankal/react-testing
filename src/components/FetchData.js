import React, { Component } from 'react'

export default class FetchData extends Component {
    constructor(props){
        super(props);

        this.state = {
            name:'alpha'
        }
    }

    handleInputChange = (e) => {
        this.setState({name:e.target.value});
    }
    render() {
        return (
            <div>
                <input type="text" value={this.state.name}/>
            </div>
        )
    }
}
