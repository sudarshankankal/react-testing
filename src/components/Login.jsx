import React, { useState } from "react";
import { Button, Card, Form } from "react-bootstrap";

export const Login = ({ onSubmit }) => {
  const [email, updateEmail] = useState("");
  const [password, updatePassword] = useState("");
  const handleLogin = (e) => {
    e.preventDefault();
    // onSubmit();
    const user = {email,password}
    return user;
  };
  return (
    <Card>
      <Card.Body>
      <Card.Header>Sign In</Card.Header>
      {/* <Card.Title>Login</Card.Title> */}
        <Form>
          <Form.Group className="mb-3" controlId="email">
            <Form.Label>Email</Form.Label>
            <input
              className="form-control"
              type="email"
              placeholder="Email"
              value={email}
              name="email"
              onChange={(e) => updateEmail(e.target.value)}
            />
          </Form.Group>

          <Form.Group className="mb-3" controlId="formBasicPassword">
            <Form.Label>Password</Form.Label>
            <input
              className="form-control"
              type="password"
              placeholder="Password"
              value={password}
              name="password"
              onChange={(e) => updatePassword(e.target.value)}
            />
          </Form.Group>

          <Button className="fullWidth" variant="info" type="submit" name="submit" onClick={(e) => handleLogin(e)}>
            Sign In
          </Button>
        </Form>
      </Card.Body>
    </Card>
  );
};
