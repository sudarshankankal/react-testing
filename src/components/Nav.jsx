import React from "react";
import { Container, Navbar } from "react-bootstrap";

function Nav() {
  return (
    <Navbar bg="light">
      <Container>
        <Navbar.Brand>React Unit Testing (Jest/Enzyme)</Navbar.Brand>
      </Container>
    </Navbar>
  );
}

export default Nav;
