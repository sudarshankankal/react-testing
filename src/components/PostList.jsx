import React, { useContext, useEffect } from "react";
import { Store } from "./StoreProvider";

const PostList = () => {
  const data = useContext(Store);

  const { postList, setPostList, setPostId, postIds } = data;

  useEffect(() => {
    fetch("https://jsonplaceholder.typicode.com/posts")
      .then((res) => res.json())
      .then((result) => setPostList(result));
  }, []);

  if (!postList.length) {
    return <div>Post's Not Found!</div>;
  }

  return (
    <div>
      <ul>
        {postList.map((post) => {
          return (
            <li
              key={post.id}
              onClick={() => setPostId([...postIds, post.id])}
              className="postItem"
            >
              {post.title}
            </li>
          );
        })}
      </ul>
    </div>
  );
};

export default PostList;
