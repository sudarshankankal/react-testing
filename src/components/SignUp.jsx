import React, { useState } from "react";
import { Button, Card, Form } from "react-bootstrap";

export const SignUp = ({ onSubmit }) => {
  const [email, setEmail] = useState("");
  const [password, setPassword] = useState("");
  const [fullName, setFullName] = useState("");
  const [phone, setPhone] = useState("");
  const handleLogin = (e) => {
    e.preventDefault();
    // onSubmit();
  };
  return (
    <Card>
      <Card.Body>
      <Card.Header>Sign Up</Card.Header>
      {/* <Card.Title>Login</Card.Title> */}
        <Form>

        <Form.Group className="mb-3" controlId="fullName">
            <Form.Label>Full Name</Form.Label>
            <input
              className="form-control"
              type="fullName"
              placeholder="Full Name"
              value={fullName}
              name="fullName"
              onChange={(e) => setFullName(e.target.value)}
            />
          </Form.Group>

          <Form.Group className="mb-3" controlId="email">
            <Form.Label>Email</Form.Label>
            <input
              className="form-control"
              type="email"
              placeholder="Email"
              value={email}
              name="email"
              onChange={(e) => setEmail(e.target.value)}
            />
          </Form.Group>

          <Form.Group className="mb-3" controlId="phone">
            <Form.Label>Phone Number</Form.Label>
            <input
              className="form-control"
              type="phone"
              placeholder="Phone Number"
              value={phone}
              name="phone"
              onChange={(e) => setPhone(e.target.value)}
            />
          </Form.Group>

          <Form.Group className="mb-3" controlId="password">
            <Form.Label>Password</Form.Label>
            <input
              className="form-control"
              type="password"
              placeholder="Password"
              value={password}
              name="password"
              onChange={(e) => setPassword(e.target.value)}
            />
          </Form.Group>

          <Button className="fullWidth" variant="info" type="submit" name="submit" onClick={(e) => handleLogin(e)}>
            Sign Up
          </Button>
        </Form>
      </Card.Body>
    </Card>
  );
};
