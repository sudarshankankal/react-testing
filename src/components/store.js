import React, { useState } from "react";

export const Store = React.createContext();

export const StoreProvider = ({ children }) => {
  const [postList, setPostList] = useState([]);
  const [postIds, setPostId] = useState([]);

  return (
    <Store.Provider value={{ postList, setPostList, postIds, setPostId }}>
      {children}
    </Store.Provider>
  );
};
