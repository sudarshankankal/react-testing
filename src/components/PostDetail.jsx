import React, { useContext } from 'react';
import { Store } from './store';

const PostDetail = () => {
    const data = useContext(Store);
    const {postIds,postList,setPostId} = data
    console.log("Post Id >>>",postIds);
    if(postIds.length === 0){
        return <div>Select Any Post</div>
    }

    


    return <div>{postList.filter((post) => postIds.includes(post.id)).map((item) => {
        return (
            <div className="postDetail">
                <div>ID: {item.id}<button onClick={() => {
                    const posts = postIds.filter(postId => postId !== item.id);
                    console.log("Remaining >>",posts)
                    setPostId(posts)
                }} style={{float:'right',color:'red'}}>Delete</button></div>
                <div>Title: {item.title}</div>
                <div>Body: {item.body}</div>
            </div>
        )
    })}</div>
}

export default PostDetail;