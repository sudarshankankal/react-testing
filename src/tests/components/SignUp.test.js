import React from "react";
import { shallow } from "enzyme";
import { SignUp } from "../../components/SignUp";

describe("Given Sign Up", () => {
  const wrapper = shallow(<SignUp />);

  test("Sign Up Form Should render properly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should render full name input", () => {
    const isPresent = wrapper.find('input[name="fullName"]').exists();
    expect(isPresent).toBe(true);
  });

  test("should render email input", () => {
    const isPresent = wrapper.find('input[name="email"]').exists();
    expect(isPresent).toBe(true);
  });

  test("should render Phone Number input", () => {
    const isPresent = wrapper.find('input[name="phone"]').exists();
    expect(isPresent).toBe(true);
  });

  test("should render password input", () => {
    const isPresent = wrapper.find('input[name="password"]').exists();
    expect(isPresent).toBe(true);
  });

  test("should render submit button", () => {
    const isPresent = wrapper.find('Button[type="submit"]').exists();
    expect(isPresent).toBe(true);
  });

  test("Default value for all the field should be empty", () => {
    const fullName = wrapper.find('input[name="fullName"]').prop("value");
    const email = wrapper.find('input[name="email"]').prop("value");
    const phone = wrapper.find('input[name="phone"]').prop("value");
    const password = wrapper.find('input[name="password"]').prop("value");

    expect(fullName).toBe("");
    expect(email).toBe("");
    expect(phone).toBe("");
    expect(password).toBe("");
  });

  test("on enter full name state should update", () => {
    wrapper.find('input[name="fullName"]').simulate("change", {
      target: {
        value: "John Doe",
      },
    });

    const name = wrapper.find('input[name="fullName"]').prop("value");
    expect(name).toBe("John Doe");
  });

  test("on enter email state should update", () => {
    wrapper.find('input[name="email"]').simulate("change", {
      target: {
        value: "alpha@email.com",
      },
    });

    const email = wrapper.find('input[name="email"]').prop("value");
    expect(email).toBe("alpha@email.com");
  });

  test("on enter Phone number state should update", () => {
    wrapper.find('input[name="phone"]').simulate("change", {
      target: {
        value: "1234567890",
      },
    });

    const phone = wrapper.find('input[name="phone"]').prop("value");
    expect(phone).toBe("1234567890");
  });

  test("on enter password state should update", () => {
    wrapper.find('input[name="password"]').simulate("change", {
      target: {
        value: "alpha@123",
      },
    });

    const password = wrapper.find('input[name="password"]').prop("value");
    expect(password).toBe("alpha@123");
  });
});
