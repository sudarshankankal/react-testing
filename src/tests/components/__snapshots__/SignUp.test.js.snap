// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given Sign Up Sign Up Form Should render properly 1`] = `
<Card
  body={false}
>
  <CardBody>
    <CardHeader>
      Sign Up
    </CardHeader>
    <Form
      inline={false}
    >
      <FormGroup
        className="mb-3"
        controlId="fullName"
      >
        <FormLabel
          column={false}
          srOnly={false}
        >
          Full Name
        </FormLabel>
        <input
          className="form-control"
          name="fullName"
          onChange={[Function]}
          placeholder="Full Name"
          type="fullName"
          value=""
        />
      </FormGroup>
      <FormGroup
        className="mb-3"
        controlId="email"
      >
        <FormLabel
          column={false}
          srOnly={false}
        >
          Email
        </FormLabel>
        <input
          className="form-control"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          type="email"
          value=""
        />
      </FormGroup>
      <FormGroup
        className="mb-3"
        controlId="phone"
      >
        <FormLabel
          column={false}
          srOnly={false}
        >
          Phone Number
        </FormLabel>
        <input
          className="form-control"
          name="phone"
          onChange={[Function]}
          placeholder="Phone Number"
          type="phone"
          value=""
        />
      </FormGroup>
      <FormGroup
        className="mb-3"
        controlId="password"
      >
        <FormLabel
          column={false}
          srOnly={false}
        >
          Password
        </FormLabel>
        <input
          className="form-control"
          name="password"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value=""
        />
      </FormGroup>
      <Button
        active={false}
        className="fullWidth"
        disabled={false}
        name="submit"
        onClick={[Function]}
        type="submit"
        variant="info"
      >
        Sign Up
      </Button>
    </Form>
  </CardBody>
</Card>
`;
